bot/exts/evergreen/latex.py

```diff
@@ -0,0 +1,75 @@
+import asyncio
+import re
+from concurrent.futures import ThreadPoolExecutor
+from io import BytesIO
+
+import discord
+import matplotlib.pyplot as plt
+from discord.ext import commands
+
+# configure fonts and colors for matplotlib
+plt.rcParams.update(
+    {
+        "font.size": 16,
+        "mathtext.fontset": "cm",  # Computer Modern font set
+        "mathtext.rm": "serif",
+        "figure.facecolor": "38383F",  # matches Discord's dark mode background color
+        "text.color": "white",
+    }
+)
+
+FORMATTED_CODE_REGEX = re.compile(
+    r"(?P<delim>(?P<block>```)|``?)"  # code delimiter: 1-3 backticks; (?P=block) only matches if it's a block
+    r"(?(block)(?:(?P<lang>[a-z]+)\n)?)"  # if we're in a block, match optional language (only letters plus newline)
+    r"(?:[ \t]*\n)*"  # any blank (empty or tabs/spaces only) lines before the code
+    r"(?P<code>.*?)"  # extract all code inside the markup
+    r"\s*"  # any more whitespace before the end of the code markup
+    r"(?P=delim)",  # match the exact same delimiter from the start again
+    re.DOTALL | re.IGNORECASE,  # "." also matches newlines, case insensitive
+)
+
+
+class Latex(commands.Cog):
+    """Renders latex."""
+
+    @staticmethod
+    def _render(text: str) -> BytesIO:
+        """Return the rendered image if latex compiles without errors, otherwise raise a BadArgument Exception."""
+        fig = plt.figure()
+        rendered_image = BytesIO()
+        fig.text(0, 1, text, horizontalalignment="left", verticalalignment="top")
+
+        try:
+            plt.savefig(rendered_image, bbox_inches="tight", dpi=600)
+        except ValueError as e:
+            raise commands.BadArgument(str(e))
+
+        rendered_image.seek(0)
+        return rendered_image
+
+    @staticmethod
+    def _prepare_input(text: str) -> str:
+        text = text.replace(r"\\", "$\n$")  # matplotlib uses \n for newlines, not \\
+
+        if match := FORMATTED_CODE_REGEX.match(text):
+            return match.group("code")
+        else:
+            return text
+
+    @commands.command()
+    async def latex(self, ctx: commands.Context, *, text: str) -> None:
+        """Renders the text in latex and sends the image."""
+        text = self._prepare_input(text)
+        async with ctx.typing():
+
+            with ThreadPoolExecutor() as pool:
+                image = await asyncio.get_running_loop().run_in_executor(
+                    pool, self._render, text
+                )
+
+            await ctx.send(file=discord.File(image, "latex.png"))
```

Comment:
You would need to `import time` to run this code.

```suggestion
        text = self._prepare_input(text)
        async with ctx.typing():
            with ThreadPoolExecutor() as pool:
                start = time.time()
                image = await asyncio.get_running_loop().run_in_executor(
                    pool, self._render, text
                )
                end = time.time()

                time_taken = end - start

            embed = discord.Embed(name="Latex Output")
            embed.set_footer(text=f"Rendering the latex image took {time_taken}s.")
            file = discord.File(image, "latex.png")
            embed.set_image(url='attachment://latex.png')

            await ctx.send(embed=embed, file=file)
```

Was messing up with the output message, and I think I came up with something good, here is the result:

### Dark Mode
![Screenshot from 2021-04-10 07-19-02](https://user-images.githubusercontent.com/69356296/114254603-fbfded00-99cd-11eb-90c4-cec2ec4f4f13.png)


### Light Mode
![Screenshot from 2021-04-10 07-19-54](https://user-images.githubusercontent.com/69356296/114254600-f86a6600-99cd-11eb-8ae2-afc33a48f4fb.png)

Reply:
Per discussion in #dev-contrib I'm not implementing this